static/app/components/dropdownMenuItemV2.tsx

```diff
@@ -0,0 +1,338 @@
+   * when `children` is also defined.
+   */
+  isSubmenu?: boolean;
+  /**
+   * Passed as the `menuTitle` prop onto the associated sub-menu (applicable
+   * if `children` is defined and `isSubmenu` is true)
+   */
+  submenuTitle?: string;
+  /**
+   * Whether to show a line divider below this menu item
+   */
+  showDividers?: boolean;
+  /*
+   * Items to be added to the left of the label
+   */
+  leadingItems?: React.ReactNode;
+  /*
+   * Whether leading items should be centered wrt/ the entire height of the
+   * menu item. If false (default), they will be centered wrt/ the first line of the
+   * label element.
+   */
+  leadingItemsSpanFullHeight?: boolean;
+  /*
+   * Items to be added to the right of the label
+   */
+  trailingItems?: React.ReactNode;
+  /*
+   * Whether trailing items should be centered wrt/ the entire height of the
+   * menu item. If false (default), they will be centered wrt/ the first line of the
+   * label element.
+   */
+  trailingItemsSpanFullHeight?: boolean;
+  /**
+   * Function to call when user selects/clicks/taps on the menu item. The
+   * item's key is passed as an argument.
+   */
+  onAction?: (key: MenuItemProps['key']) => void;
+  /**
+   * Sub-items that are nested inside this item. By default, sub-items are
+   * rendered collectively as menu sections inside the current menu. If
+   * `isSubmenu` is true, then they will be rendered together in a sub-menu.
+   */
+  children?: MenuItemProps[];
+};
+
+type Props = {
+  /**
+   * Node representation (from @react-aria) of the item
+   */
+  node: Node<MenuItemProps>;
+  /**
+   * Whether this is the last node in the collection
+   */
+  isLastNode: boolean;
+  /**
+   * Tree state (from @react-stately) inherited from parent menu
+   */
+  state: TreeState<MenuItemProps>;
+  /**
+   * Used to close the menu when needed (e.g. when the item is
+   * clicked/selected)
+   */
+  onClose: () => void;
+  /**
+   * Whether to close the menu when an item has been clicked/selected
+   */
+  closeOnSelect: boolean;
+  /**
+   * Whether this is a trigger button (displayed as a normal menu item) for a
+   * submenu
+   */
+  isSubmenuTrigger?: boolean;
+  /**
+   * Tag name for item wrapper
+   */
+  renderAs?: React.ElementType;
+};
+
+/**
+ * A menu item with a label, optional details, leading and trailing elements.
+ * Can also be used as a trigger button for a submenu. See:
+ * https://react-spectrum.adobe.com/react-aria/useMenu.html
+ */
+const MenuItem = forwardRef<React.RefObject<HTMLLIElement>, Props>(
+  (
+    {
+      node,
+      isLastNode,
+      state,
+      onClose,
+      closeOnSelect,
+      isSubmenuTrigger = false,
+      renderAs = 'li' as React.ElementType,
+      ...submenuTriggerProps
+    },
+    submenuTriggerRef
+  ) => {
+    const [isHovering, setIsHovering] = useState(false);
+    const ref = (submenuTriggerRef ?? useRef(null)) as React.RefObject<HTMLLIElement>;
```

Comment:
```suggestion
  const ref = submenuTriggerRef ?? useRef<HTMLLIElement>(null)
```

Reply:
Had to add a type assertion here because otherwise TypeScript wouldn't understand that if `submenuTriggerRef` is `null` then it would fall back to an internal ref object (`useRef(null)`):

<img width="909" alt="Screen Shot 2022-01-31 at 9 26 58 AM" src="https://user-images.githubusercontent.com/44172267/151842756-be27635f-a6fe-44b5-bc71-7404b81398ad.png">